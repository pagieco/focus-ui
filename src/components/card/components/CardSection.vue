<script>

import { filterClassList } from '../../../utilities';

export default {
  name: 'focus-card-section',

  props: {
    /**
     * Title content for the card.
     */
    title: {
      type: String,
    },

    /**
     * A less prominent card.
     */
    subdued: {
      type: Boolean,
      default: false,
    },

    /**
     * Minimize the cards' padding.
     */
    shallow: {
      type: Boolean,
      default: false,
    },
  },

  computed: {
    classList() {
      return filterClassList([
        'card__section',
        { 'card__section--subdued': this.subdued },
        { 'card__section--shallow': this.shallow },
      ]);
    },
  },
};

</script>

<template>
  <div :class="classList">
    <header v-if="title">
      <h3 :aria-label="title">{{ title }}</h3>
    </header>

    <slot />
  </div>
</template>
