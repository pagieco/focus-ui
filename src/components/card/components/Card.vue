<script>

import CardSection from './CardSection.vue';
import { filterClassList } from '../../../utilities';

export default {
  name: 'focus-card',

  components: { CardSection },

  props: {
    /**
     * Indicates that the card is interactive.
     */
    interactive: {
      type: Boolean,
      default: false,
    },

    /**
     * Auto wrap content in section.
     */
    sectioned: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    classList() {
      return filterClassList([
        'card',
        { 'card--interactive': this.interactive },
      ]);
    },
  },

  render(h) {
    return h('div', { class: this.classList }, [
      h(this.sectioned ? CardSection : 'div', [
        this.$slots.default,
      ]),
    ]);
  },
};

</script>
