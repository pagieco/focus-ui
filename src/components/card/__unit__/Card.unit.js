import { shallowMount } from '@vue/test-utils';
import Card from '../components/Card.vue';
import CardSection from '../components/CardSection.vue';

const shallowWrap = propsData => shallowMount(Card, { propsData });

describe('Card', () => {
  it('is a valid vue component', () => {
    expect(Card).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowMount(Card).element).toMatchSnapshot();
  });

  it('uses the interactive prop', () => {
    const wrapper = shallowWrap({ interactive: true });
    const expectedClass = 'card--interactive';

    expect(wrapper.classes()).toContain(expectedClass);

    wrapper.setProps({ interactive: false });

    expect(wrapper.classes()).not.toContain(expectedClass);
  });

  it('uses the sectioned prop', () => {
    const wrapper = shallowMount(Card);

    expect(wrapper.contains(CardSection)).toBeTruthy();

    wrapper.setProps({ sectioned: false });

    expect(wrapper.contains(CardSection)).toBeFalsy();
  });

  it('renders child components', () => {
    const wrapper = shallowMount(Card, {
      slots: {
        default: '<div>Slot content</div>',
      },
    });

    expect(wrapper.text()).toContain('Slot content');
  });
});
