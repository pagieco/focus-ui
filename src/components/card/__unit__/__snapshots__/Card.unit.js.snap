// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<div
  class="card"
>
  <focus-card-section-stub />
</div>
`;
